src/common/queries/useBlockByHash.ts

```diff
@@ -2,47 +2,50 @@ import { UseSuspenseQueryOptions, useQuery, useSuspenseQuery } from '@tanstack/r
 
 import { Block } from '@stacks/stacks-blockchain-api-types';
 
-import { useApi } from '../api/useApi';
-import { useGlobalContext } from '../context/useGlobalContext';
+import { getErrorMessage } from '../../api/getErrorMessage';
+import { useApiClient } from '../../api/useApiClient';
 
 const BLOCK_QUERY_KEY = 'block';
 
 export function useBlockByHash(
```

Comment:
So we need this if we have the blockbyheightorhash endpoint?

Reply:
mind elaborating on this one? I did a direct refactor with the same endpoint that's being used.

Reply:
I meant we have this hook, useSuspenseBlockByHeightOrHash, which uses a v2 endpoint '/extended/v2/blocks/{height_or_hash}', versus the v1 endpoint '/extended/v1/block/{hash}', used by useBlockByHash. Can we just delete useBlockByHash, since it's not as functional as useSuspenseBlockByHeightOrHash, and we don't need both